### PR TITLE
254 physics geometry plane uses vector3 but is 2d

### LINKED ISSUE #254: Physics geometry plane uses Vector3 but is 2d
Size variable is 3d but z dimension is completely ignored. make it some vector2d.

### PR DESCRIPTION
**Pull request by oliverweissl**
Vector2 class added in simulation package.
Behaves as the Vector3 class in pyrr.

 closes #254

### CODE REVIEW

File: simulation/revolve2/simulation/custom_vector/object/vector2.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+from numbers import Number
+from typing import Any
+
+import numpy as np
+from pyrr.objects.base import BaseMatrix33, BaseVector, NpProxy
+
+from .. import vector2
+
+
+class BaseVector2(BaseVector):  # type:ignore
```
Comment (surgura) [inline simulation/revolve2/simulation/custom_vector/object/vector2.py:13]: What is the reason this class exist. why does vector2 not directly inherit from BaseVector?

File: simulation/revolve2/simulation/custom_vector/object/vector2.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    y = NpProxy(1)
+    #: The X,Y values of this Vector as a numpy.ndarray.
+    xy = NpProxy([0, 1])
+
+    ########################
+    # Creation
+    @classmethod
+    def from_vector3(cls, vector: Any, dtype: Any = None) -> tuple[Vector2, float]:
+        """
+        Create a Vector2 from a Vector3.
```
[the remark this answers is not in the document]
Comment (surgura) [reply simulation/revolve2/simulation/custom_vector/object/vector2.py:37]: I think conversion from 3d to 2d is not intuitive and is something that the user will have to explicitly do themselves.

File: simulation/revolve2/simulation/custom_vector/object/vector2.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
+"""A representation of a 2d Vector."""
```
Comment (surgura) [inline simulation/revolve2/simulation/custom_vector/object/vector2.py:1]: I prefer to drop the 'object' directory.
I think `from revolve2.simulation.vector2 import Vector2, unit, ...etc`
would be nice

File: simulation/revolve2/simulation/custom_vector/object/vector2.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (oliverweissl) [reply simulation/revolve2/simulation/custom_vector/object/vector2.py:37]: this function i created from the pyrr function from_vector4, where it drops the last dimension to get a flatter vector.
I can make the description better or remove it completly. I just wanted to mimic pyrr`s vectors as closely as possible.

[the remark this answers is not in the document]
Comment (surgura) [reply simulation/revolve2/simulation/custom_vector/object/vector2.py]: Ok I understand. Then I think that is a weird feature of pyrr that we should not port :) Imo pretty arbitrary to just drop the last dimension.

**Comment by surgura** [comment]
if you rebase to dev i'll do a final review :+1:

**Comment by oliverweissl** [comment]
index is used so i left it in

File: simulation/revolve2/simulation/vector2/vector2.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+        if type(other) in self.__NMB:
+            return Vector2(super(Vector2, self).__sub__(other))
+        elif type(other) in self.__VCT:
+            return Vector2(super(Vector2, self).__sub__(other))
+        else:
+            self._unsupported_type("subtract", other)
+
+    def __mul__(self, other: Any) -> Vector2:  # type:ignore
+        """
+        Multiply the existing Vector2.
```
[the remark this answers is not in the document]
Comment (oliverweissl) [reply simulation/revolve2/simulation/vector2/vector2.py:86]: i removed the vector multiplication,  its just with numbers now

File: simulation/revolve2/simulation/vector2/vector2aux.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    Create a Vector2 from a 3x3 Matrix.
+
+    :param mat: The 3x3 Matrix.
+    :param dtype: The data-type.
+    :return: The Vector2.
+    """
+    return np.array(mat[2, :2], dtype=dtype)
+
+
+class index:
```
[the remark this answers is not in the document]
Comment (oliverweissl) [reply simulation/revolve2/simulation/vector2/vector2aux.py:65]: the index is actually used :D

File: simulators/mujoco/mypy.ini [hunk 6]
```diff
@@ -13,3 +13,6 @@ ignore_missing_imports = True
 
 [mypy-dm_control.*]
 ignore_missing_imports = True
+
+[mypy-cv2.*]
+ignore_missing_imports = True
```
[the remark this answers is not in the document]
Comment (oliverweissl) [reply simulators/mujoco/mypy.ini:15]: theres no line in my document after 15

File: simulators/mujoco/mypy.ini [hunk 7]
```diff
@@ -12,4 +12,4 @@ ignore_missing_imports = True
 ignore_missing_imports = True
 
 [mypy-dm_control.*]
-ignore_missing_imports = True
```
Comment (surgura) [inline simulators/mujoco/mypy.ini:15]: This change is not relevant to this issue

**Review by surgura — COMMENTED** [review]
Accept besides that final thing

**Review by surgura — APPROVED** [review]

**Pull request merged**